packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -478,7 +478,7 @@ class AndroidDevice extends Device {
 
   @override
   Future<bool> uninstallApp(
-    AndroidApk app, {
+    ApplicationPackage app, {
```

Comment:
why not covariant AndroidApk here?

Reply:
Because `covariant AndroidApk` would turns off the static analysis for this type, so if the base class switched to `ApplicationPackage?` say, the analyzer wouldn't complain and we'd be back where we started.

Reply:
yup, I forgot